File: c++/Empleado.cc
#include "Empleado.h"
#include "Persona.h"
#include "Jefe.h"
#include <iostream>

using namespace std;

Empleado::Empleado(int e, double s) {
  edad = e;
  sueldo = s;
}

void Empleado::print() {
  cout << "Soy trabajador y necesito " << sueldo << "\n";
}

int main() {
  Persona* p1 = new Persona(24);
  p1->print();
  Persona* p2 = new Empleado(34, 3000.50);
  p2->print();
  Jefe* j = new Jefe();
  j->ordenar();
  j->preguntar();
}

File: java/Empleado.java
public class Empleado extends Persona {
  double sueldo;
  
  public Empleado(String n, int e, double c) {
    super(n, e);
    this.sueldo = c;
  }

  public void print() {
    System.out.println("A trabajar, " + nombre + " hay que ganar esos " + sueldo + " soles.");
  }

  public static void main(String[] args) {
    //Herencia de Clase:
    Persona p = new Persona("Juan", 12);
    p.print();
    Persona emp = new Empleado("Alex", 34, 3100.40);
    emp.print();

    //Herencia de tipo (Interface)
    Jefe jefe = new Jefe();
    jefe.preguntar();
    jefe.ordenar();
  }
}